src/riak_kv_tictacaae_repairs.erl

```diff
@@ -0,0 +1,376 @@
+                            -> {false|bucket|modtime, aae_exchange:filters()}.
+analyse_repairs(_RepairList, _MaxRepairs, _EnableKeyRange, false) ->
+    {false, none};
+analyse_repairs(RepairList, MaxRepairs, EnableKeyRange, _) ->
+    case lists:reverse(lists:keysort(2, RepairList)) of
+        [Candidate|_Rest] ->
+            Threshold = MaxRepairs div 3,
+            case Candidate of
+                {B, CandCount, KL, MTL} when CandCount > (2 * Threshold) ->
+                    [FirstKey|RestKeys] = lists:sort(KL),
+                    LastKey = lists:last(RestKeys),
+                    KeyRange =
+                        case EnableKeyRange of 
+                            true -> {FirstKey, LastKey};
+                            false -> all
+                        end,
+                    {bucket,
+                        {filter,
+                            B, KeyRange, large, all,
+                            get_modified_range(MTL),
+                            pre_hash}};
+                _ ->
+                    FoldFun = fun({_B, _C, _KL, MTL}, Acc) -> Acc ++ MTL end,
+                    MTL = lists:sort(lists:foldl(FoldFun, [], RepairList)),
+                    case length(MTL) of
+                        CandCount when CandCount > Threshold ->
+                            {modtime, 
+                                {filter,
+                                    all, all, large, all,
+                                    get_modified_range(MTL),
+                                    pre_hash}};
+                        _ ->
+                            {false, none}
+                    end
+            end;
+        [] ->
+            {false, none}
+    end.
+            
+
+-spec get_modified_range(list(calendar:datetime()))
+                            -> {pos_integer(), pos_integer()}.
+get_modified_range(ModifiedDateTimeList) ->
+    [FirstDate|RestDates] = lists:sort(ModifiedDateTimeList),
+    HighDate = lists:last(RestDates),
+    EpochTime =
+        calendar:datetime_to_gregorian_seconds({{1970,1,1},{0,0,0}}),
+    LowTS = 
+        calendar:datetime_to_gregorian_seconds(FirstDate) - EpochTime,
+    HighTS =
+        calendar:datetime_to_gregorian_seconds(HighDate) - EpochTime,
+    {LowTS - 1, HighTS + 1}.
+
+%% @doc Look at the last modified time on each clock
+%% Further repairs will only on those prompted in the clean case
+%% when the last modified time is different between the clocks - i.e.
+%% one clock has an identifiably higher last modified time which can be
+%% considered a likely timestamp of an incident
+-spec analyse_repair({{riak_object:bucket(), riak_object:key()},
+                        {vclock:vclock(), vclock:vclock()}},
+                        repair_list()) -> repair_list().
+analyse_repair({{B, K}, {BlueClock, PinkClock}}, ByBucketAcc) ->
+    BlueTime = last_modified(BlueClock),
+    PinkTime = last_modified(PinkClock),
+    ModTime = 
+        case {BlueTime, PinkTime} of
```

Comment:
This probably would be clearer in an ìf`, the matching is a bit weird here.

Reply:
Change made